Sources/AdminPanel/Controllers/ResetController.swift

```diff
@@ -0,0 +1,110 @@
+import Leaf
+import Reset
+import Submissions
+import Sugar
+import Vapor
+
+internal final class ResetController
+    <U: AdminPanelUserType>: ResetControllerType
+{
+    internal func renderResetPasswordRequestForm(_ req: Request) throws -> Future<Response> {
+        let adminPanelConfig: AdminPanelConfig<U> = try req.make()
+        try req.populateFields(U.RequestReset.self)
+        return try req.privateContainer
+            .make(LeafRenderer.self)
+            .render(adminPanelConfig.views.login.requestResetPassword)
+            .encode(for: req)
+    }
+
+    internal func resetPasswordRequest(_ req: Request) throws -> Future<Response> {
+        let resetConfig: ResetConfig<U> = try req.make()
+        let adminPanelConfig: AdminPanelConfig<U> = try req.make()
+        let submission = try req.content.decode(U.RequestReset.Submission.self)
```

Comment:
Guidelines ... ;)

Reply:
I'm afraid I cannot do this here. The signature looks like this:
```swift
public func decode<D>(_ content: D.Type, maxSize: Int = 65_536) throws -> Future<D> where D: Decodable
```
and there's no overload with a default value on `content`.

Reply:
ahh yeah that's a no-go then. Would be nice to detect all these occurrences in Vapor's code and fix them.